src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosBlockSignatureRepresentationRule.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Check that the block signature for a POS block is in the canonical format.
+    /// </summary>
+    public class PosBlockSignatureRepresentationRule : PartialValidationConsensusRule
+    {
+        /// <inheritdoc />
+        public override Task RunAsync(RuleContext context)
+        {
+            if (!PosBlockValidator.IsCanonicalBlockSignature((PosBlock)context.ValidationContext.BlockToValidate, false))
+            {
+                // In stratisX CANONICAL_BLOCK_SIG_VERSION (version.h) is defined as 70000, the same as ProtocolVersion.ALT_PROTOCOL_VERSION.
+                if (this.Parent.NodeSettings.ProtocolVersion >= ProtocolVersion.ALT_PROTOCOL_VERSION)
```

Comment:
Can't you add CANONICAL_BLOCK_SIG_VERSION to ProtocolVersion, with a value 7000 then no need to add the comment and CANONICAL_BLOCK_SIG_VERSION could be used elsewhere, if need be ?

Reply:
you dont need settings we plan to put protocol version on the network class (most likely)

Reply:
you also have a bug here, in stratisx we throw regardless of protocol version (PV is only used to determine if we need to ban the peer)

Reply:
to avoid passing settings everywhere I suggest we even just use numbers to determine banning

Reply:
also you are checking the protocol version of our node not the peers